module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php

```diff
@@ -120,54 +120,129 @@ public function getAccessRestrictions()
      */
     public function getAllSubjectHeadings($extended = false)
     {
-        // This is all the collected data:
-        $retval = [];
+        if (($this->mainConfig->Record->marcSubjectHeadingsSort ?? '') === 'field') {
+            $returnValues = $this->getAllSubjectHeadingsNumericalOrder($extended);
+        } else {
+            // Default | value === 'marc'
```

Comment:
This should be 'record' rather than 'marc' now if we want to match what the configuration file says. And maybe we should rename the method to getAllSubjectHeadingsRecordOrder to match. Or, if you prefer, we could change the config file to use "marc" instead of "record" as the config setting. I think either one is potentially valid. You could also change "field" to "numerical" if you think that's more clear. I don't have strong feelings about any of this, except that I think it would be helpful if we aim for consistency in all places.

Reply:
I agree, done